### ISSUE TITLE
HNC: support dry-run

### ISSUE THREAD

**Issue by adrianludwin** [kubernetes-sigs/multi-tenancy#1027]
Per a comment on Slack, when HNC is installed you can no longer run:

```
kubectl run nginx --generator=run-pod/v1 --image=nginx --dry-run=server
```

This is likely a webhook config problem.

_Update Aug 26: requires K8s 1.18 (KIND is fine) and `--dry-run=server`_

**Comment by adrianludwin** [comment]
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#side-effects

**Comment by adrianludwin** [comment]
This should be easy enough to pull into 0.5.2.

**Comment by adrianludwin** [comment]
Hmm, this appears to be working for me. Not sure why.

**Comment by adrianludwin** [comment]
Not only can I create pods with dry-run, but also namespaces and `HierarchyConfigurations`. This despite the fact that the webhooks all have `sideEffect: Unknown`. I don't understand why this is working.

**Comment by adrianludwin** [comment]
Turns out I've been testing on 1.16, while this feature was only enabled in 1.18. I'll have another go tomorrow.

**Comment by adrianludwin** [comment]
Reproduced successfully

**Comment by adrianludwin** [comment]
Fixed with a simple kustomization patch

**Comment by adrianludwin** [comment]
Fixed in 0.5.2

/close
